data-source/src/main/scala/tech/ytsaurus/spyt/format/YtDynamicTableWriter.scala

```diff
@@ -28,10 +26,9 @@ class YtDynamicTableWriter(richPath: YPathEnriched,
 
   override val path: String = richPath.toStringPath
 
-  private val schemaHint = options.ytConf(WriteSchemaHint)
-  private val typeV3Format = options.ytConf(WriteTypeV3)
+  private val writeSchemaConverter = new WriteSchemaConverter(options)
```

Comment:
I think you should revisit all occurences of WriteSchemaConverter creation, because in come cases you create it via new, or via apply, sometimes you explicitly specify default values of arguments, sometimes not, and so on